File: Codewars/Persistent Bugger.c
typedef struct{
  int a;
  int b;
}data;

data len(n){
  int c = 0, _m = 1;
  while(n > 0)
    _m *= n%10, n /= 10, c++;
  return (data){c==0 ? 1 : c, _m};
}

int persistence(int n) {
  int c = 0;
  data counter = len(n);
  while(counter.a != 1)
    counter = len(counter.b), c++;
  return c;
}

File: Codewars/Persistent Bugger.py
def split(s):
    return [int(item) for item in str(s)];

def m(s):
    _m = 1;
    for item in s:
        _m *= item;
    return _m;

def persistence(n):
    c = 1;
    s = split(n);
    if(len(s)==1):
        return 0;
    _m = m(s);
    while(len(str(_m)) != 1):
        s = split(_m);
        _m = m(s);
        c += 1;
    return c;
